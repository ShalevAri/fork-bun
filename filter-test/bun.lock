{
  "lockfileVersion": 1,
  "workspaces": {
    "": {
      "name": "root",
      "dependencies": {
        "lodash": "^4.17.21",
      },
    },
    "packages/pkg1": {
      "name": "pkg1",
      "version": "0.0.1",
    },
    "packages/pkg2": {
      "name": "pkg2",
      "version": "0.0.1",
    },
  },
  "packages": {
    "lodash": ["lodash@4.17.21", "", {}, "sha512-v2kDEe57lecTulaDIuNTPy3Ry4gLGJ6Z1O3vE1krgXZNrsQ+LFTGHVxVjcXPs17LhbZVGedAJv8XZ1tvj5FvSg=="],

    "pkg1": ["pkg1@workspace:packages/pkg1"],

    "pkg2": ["pkg2@workspace:packages/pkg2"],
  }
}
