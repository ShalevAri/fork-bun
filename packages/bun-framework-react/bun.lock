{
  "lockfileVersion": 1,
  "workspaces": {
    "": {
      "dependencies": {
        "react": "19.2.0-canary-67a44bcd-20250915",
        "react-dom": "19.2.0-canary-67a44bcd-20250915",
        "react-refresh": "^0.17.0",
        "react-server-dom-bun": "^0.0.0-experimental-603e6108-20241029",
      },
      "devDependencies": {
        "@types/react": "^19.1.13",
        "@types/react-dom": "^19.1.9",
      },
    },
  },
  "packages": {
    "@types/react": ["@types/react@19.1.13", "", { "dependencies": { "csstype": "^3.0.2" } }, "sha512-hHkbU/eoO3EG5/MZkuFSKmYqPbSVk5byPFa3e7y/8TybHiLMACgI8seVYlicwk7H5K/rI2px9xrQp/C+AUDTiQ=="],

    "@types/react-dom": ["@types/react-dom@19.1.9", "", { "peerDependencies": { "@types/react": "^19.0.0" } }, "sha512-qXRuZaOsAdXKFyOhRBg6Lqqc0yay13vN7KrIg4L7N4aaHN68ma9OK3NE1BoDFgFOTfM7zg+3/8+2n8rLUH3OKQ=="],

    "csstype": ["csstype@3.1.3", "", {}, "sha512-M1uQkMl8rQK/szD0LNhtqxIPLpimGm8sOBwU7lLnCpSbTyY3yeU1Vc7l4KT5zT4s/yOxHH5O7tIuuLOCnLADRw=="],

    "neo-async": ["neo-async@2.6.2", "", {}, "sha512-Yd3UES5mWCSqR+qNT93S3UoYUkqAZ9lLg8a7g9rimsWmYGK8cVToA4/sF3RrshdyV3sAGMXVUmpMYOw+dLpOuw=="],

    "react": ["react@19.2.0-canary-67a44bcd-20250915", "", {}, "sha512-93GXCSAJhSkIpC0Z8adssVHP/MIgmufZ0eFBBP3TUWuLYNFk7Pv8Sal6UZ/UGVeX+KZPSu95vSR5guUvqtGrDQ=="],

    "react-dom": ["react-dom@19.2.0-canary-67a44bcd-20250915", "", { "dependencies": { "scheduler": "0.27.0-canary-67a44bcd-20250915" }, "peerDependencies": { "react": "19.2.0-canary-67a44bcd-20250915" } }, "sha512-s+aP2nMr/wKdIrf0Z/fyYObl8IyMXbaW+ujekxG7hKtssuWjOJ2OhS9bg34/DFm1OaMczKKyW2Crv0BYRUlJfA=="],

    "react-refresh": ["react-refresh@0.17.0", "", {}, "sha512-z6F7K9bV85EfseRCp2bzrpyQ0Gkw1uLoCel9XBVWPg/TjRj94SkJzUTGfOa4bs7iJvBWtQG0Wq7wnI0syw3EBQ=="],

    "react-server-dom-bun": ["react-server-dom-bun@0.0.0-experimental-603e6108-20241029", "", { "dependencies": { "neo-async": "^2.6.1" } }, "sha512-FfteCHlOgJSnDJRatgIkIU74jQQ9M1+fH2e6kfY9Sibu8FAWEUjgApKQPDfiXgjrkY7w0ITQu0b2FezC0eGzCw=="],

    "scheduler": ["scheduler@0.27.0-canary-67a44bcd-20250915", "", {}, "sha512-smZ2mJWYQBKEpnEHOavM3YwhduC6pD4esZTefLPwQok/Ca65W/tWDHRHJhFHYi+A8ottJJA8G6tXfhA2wuGH0A=="],
  }
}
