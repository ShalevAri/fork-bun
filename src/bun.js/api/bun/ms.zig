const std = @import("std");
const bun = @import("bun");
const jsc = bun.jsc;
const JSValue = jsc.JSValue;
const JSGlobalObject = jsc.JSGlobalObject;

const MS_PER_SECOND = 1000.0;
const MS_PER_MINUTE = MS_PER_SECOND * 60.0;
const MS_PER_HOUR = MS_PER_MINUTE * 60.0;
const MS_PER_DAY = MS_PER_HOUR * 24.0;
const MS_PER_WEEK = MS_PER_DAY * 7.0;
const MS_PER_YEAR = MS_PER_DAY * 365.25;

/// Parse a time string like "2d", "1.5h", "5m" to milliseconds
pub fn parse(input: []const u8) ?f64 {
    if (input.len == 0 or input.len > 100) return null;

    var i: usize = 0;
    var negative = false;

    // Skip leading whitespace
    while (i < input.len and std.ascii.isWhitespace(input[i])) : (i += 1) {}
    if (i >= input.len) return null;

    // Check for negative sign
    if (input[i] == '-') {
        negative = true;
        i += 1;
    }

    // Parse number
    const start = i;
    var has_digits = false;
    var has_dot = false;

    while (i < input.len) {
        const c = input[i];
        if (std.ascii.isDigit(c)) {
            has_digits = true;
            i += 1;
        } else if (c == '.' and !has_dot) {
            has_dot = true;
            i += 1;
        } else if (std.ascii.isWhitespace(c) or std.ascii.isAlphabetic(c)) {
            break;
        } else {
            return null;
        }
    }

    if (!has_digits) return null;

    const number_str = input[start..i];
    var value = std.fmt.parseFloat(f64, number_str) catch return null;

    if (negative) value = -value;

    // Skip whitespace after number
    while (i < input.len and std.ascii.isWhitespace(input[i])) : (i += 1) {}

    // Get unit (rest of string, excluding trailing whitespace)
    var unit_end = input.len;
    while (unit_end > i and std.ascii.isWhitespace(input[unit_end - 1])) : (unit_end -= 1) {}
    const unit = input[i..unit_end];

    // Default to milliseconds if no unit
    if (unit.len == 0) return value;

    // Match unit (case-insensitive)
    return if (getMultiplier(unit)) |m| value * m else null;
}

fn getMultiplier(unit: []const u8) ?f64 {
    // Years
    if (std.ascii.eqlIgnoreCase(unit, "years") or std.ascii.eqlIgnoreCase(unit, "year") or
        std.ascii.eqlIgnoreCase(unit, "yrs") or std.ascii.eqlIgnoreCase(unit, "yr") or
        std.ascii.eqlIgnoreCase(unit, "y"))
    {
        return MS_PER_YEAR;
    }

    // Weeks
    if (std.ascii.eqlIgnoreCase(unit, "weeks") or std.ascii.eqlIgnoreCase(unit, "week") or
        std.ascii.eqlIgnoreCase(unit, "w"))
    {
        return MS_PER_WEEK;
    }

    // Days
    if (std.ascii.eqlIgnoreCase(unit, "days") or std.ascii.eqlIgnoreCase(unit, "day") or
        std.ascii.eqlIgnoreCase(unit, "d"))
    {
        return MS_PER_DAY;
    }

    // Hours
    if (std.ascii.eqlIgnoreCase(unit, "hours") or std.ascii.eqlIgnoreCase(unit, "hour") or
        std.ascii.eqlIgnoreCase(unit, "hrs") or std.ascii.eqlIgnoreCase(unit, "hr") or
        std.ascii.eqlIgnoreCase(unit, "h"))
    {
        return MS_PER_HOUR;
    }

    // Minutes
    if (std.ascii.eqlIgnoreCase(unit, "minutes") or std.ascii.eqlIgnoreCase(unit, "minute") or
        std.ascii.eqlIgnoreCase(unit, "mins") or std.ascii.eqlIgnoreCase(unit, "min") or
        std.ascii.eqlIgnoreCase(unit, "m"))
    {
        return MS_PER_MINUTE;
    }

    // Seconds
    if (std.ascii.eqlIgnoreCase(unit, "seconds") or std.ascii.eqlIgnoreCase(unit, "second") or
        std.ascii.eqlIgnoreCase(unit, "secs") or std.ascii.eqlIgnoreCase(unit, "sec") or
        std.ascii.eqlIgnoreCase(unit, "s"))
    {
        return MS_PER_SECOND;
    }

    // Milliseconds
    if (std.ascii.eqlIgnoreCase(unit, "milliseconds") or std.ascii.eqlIgnoreCase(unit, "millisecond") or
        std.ascii.eqlIgnoreCase(unit, "msecs") or std.ascii.eqlIgnoreCase(unit, "msec") or
        std.ascii.eqlIgnoreCase(unit, "ms"))
    {
        return 1.0;
    }

    return null;
}


/// Format milliseconds to a human-readable string
pub fn format(allocator: std.mem.Allocator, ms: f64, long: bool) ![]const u8 {
    const abs_ms = @abs(ms);

    if (abs_ms >= MS_PER_DAY) {
        const days = @round(ms / MS_PER_DAY);
        const days_int = @as(i64, @intFromFloat(days));
        if (long) {
            const plural = abs_ms >= MS_PER_DAY * 1.5;
            return std.fmt.allocPrint(allocator, "{d} day{s}", .{ days_int, if (plural) "s" else "" });
        }
        return std.fmt.allocPrint(allocator, "{d}d", .{days_int});
    }

    if (abs_ms >= MS_PER_HOUR) {
        const hours = @round(ms / MS_PER_HOUR);
        const hours_int = @as(i64, @intFromFloat(hours));
        if (long) {
            const plural = abs_ms >= MS_PER_HOUR * 1.5;
            return std.fmt.allocPrint(allocator, "{d} hour{s}", .{ hours_int, if (plural) "s" else "" });
        }
        return std.fmt.allocPrint(allocator, "{d}h", .{hours_int});
    }

    if (abs_ms >= MS_PER_MINUTE) {
        const minutes = @round(ms / MS_PER_MINUTE);
        const minutes_int = @as(i64, @intFromFloat(minutes));
        if (long) {
            const plural = abs_ms >= MS_PER_MINUTE * 1.5;
            return std.fmt.allocPrint(allocator, "{d} minute{s}", .{ minutes_int, if (plural) "s" else "" });
        }
        return std.fmt.allocPrint(allocator, "{d}m", .{minutes_int});
    }

    if (abs_ms >= MS_PER_SECOND) {
        const seconds = @round(ms / MS_PER_SECOND);
        const seconds_int = @as(i64, @intFromFloat(seconds));
        if (long) {
            const plural = abs_ms >= MS_PER_SECOND * 1.5;
            return std.fmt.allocPrint(allocator, "{d} second{s}", .{ seconds_int, if (plural) "s" else "" });
        }
        return std.fmt.allocPrint(allocator, "{d}s", .{seconds_int});
    }

    const ms_int = @as(i64, @intFromFloat(ms));
    if (long) {
        return std.fmt.allocPrint(allocator, "{d} ms", .{ms_int});
    }
    return std.fmt.allocPrint(allocator, "{d}ms", .{ms_int});
}

/// JavaScript function: Bun.ms(value, options?)
pub fn jsFunction(
    globalThis: *JSGlobalObject,
    callframe: *jsc.CallFrame,
) bun.JSError!jsc.JSValue {
    const args = callframe.arguments_old(2);
    if (args.len == 0) return .js_undefined;

    const input = args.ptr[0];

    // If input is a number, format it to a string
    if (input.isNumber()) {
        const ms_value = input.asNumber();

        if (std.math.isNan(ms_value) or std.math.isInf(ms_value)) {
            return globalThis.throwInvalidArguments("Value must be a finite number", .{});
        }

        // Check for options argument
        var long = false;
        if (args.len > 1 and args.ptr[1].isObject()) {
            const options = args.ptr[1];
            if (try options.get(globalThis, "long")) |long_value| {
                long = long_value.toBoolean();
            }
        }

        const result = format(bun.default_allocator, ms_value, long) catch {
            return globalThis.throwOutOfMemory();
        };
        defer bun.default_allocator.free(result);

        return bun.String.fromBytes(result).toJS(globalThis);
    }

    // If input is a string, parse it to milliseconds
    if (input.isString()) {
        const str = try input.getZigString(globalThis);
        const slice = str.toSlice(bun.default_allocator);
        defer slice.deinit();

        const result = parse(slice.slice()) orelse return .js_undefined;
        return JSValue.jsNumber(result);
    }

    // Invalid input type
    return .js_undefined;
}
