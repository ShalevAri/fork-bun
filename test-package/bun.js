export const value = "from-bun";
export const message = "This is from the Bun-specific export!";