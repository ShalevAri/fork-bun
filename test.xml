<root><message>Hello XML\!</message></root>
