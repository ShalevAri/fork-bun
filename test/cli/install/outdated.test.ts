import { test, describe, expect, beforeAll, afterAll } from "bun:test";
import { spawn, write } from "bun";
import { join } from "path";
import { VerdaccioRegistry, bunExe, bunEnv as env, assertManifestsPopulated, runBunInstall } from "harness";

let registry = new VerdaccioRegistry();

beforeAll(async () => {
  await registry.start();
});
afterAll(() => {
  registry.stop();
});

describe("outdated", () => {
  const edgeCaseTests = [
    {
      description: "normal dep, smaller than column title",
      packageJson: {
        dependencies: {
          "no-deps": "1.0.0",
        },
      },
    },
    {
      description: "normal dep, larger than column title",
      packageJson: {
        dependencies: {
          "prereleases-1": "1.0.0-future.1",
        },
      },
    },
    {
      description: "dev dep, smaller than column title",
      packageJson: {
        devDependencies: {
          "no-deps": "1.0.0",
        },
      },
    },
    {
      description: "dev dep, larger than column title",
      packageJson: {
        devDependencies: {
          "prereleases-1": "1.0.0-future.1",
        },
      },
    },
    {
      description: "peer dep, smaller than column title",
      packageJson: {
        peerDependencies: {
          "no-deps": "1.0.0",
        },
      },
    },
    {
      description: "peer dep, larger than column title",
      packageJson: {
        peerDependencies: {
          "prereleases-1": "1.0.0-future.1",
        },
      },
    },
    {
      description: "optional dep, smaller than column title",
      packageJson: {
        optionalDependencies: {
          "no-deps": "1.0.0",
        },
      },
    },
    {
      description: "optional dep, larger than column title",
      packageJson: {
        optionalDependencies: {
          "prereleases-1": "1.0.0-future.1",
        },
      },
    },
  ];

  for (const { description, packageJson } of edgeCaseTests) {
    test(description, async () => {
      const { packageDir } = await registry.createTestDir();
      await write(join(packageDir, "package.json"), JSON.stringify(packageJson));
      await runBunInstall(env, packageDir);
      assertManifestsPopulated(join(packageDir, ".bun-cache"), registry.registryUrl());

      const testEnv = { ...env, FORCE_COLOR: "1" };
      const { stdout, stderr, exited } = spawn({
        cmd: [bunExe(), "outdated"],
        cwd: packageDir,
        stdout: "pipe",
        stderr: "pipe",
        env: testEnv,
      });

      expect(await exited).toBe(0);

      const err = await Bun.readableStreamToText(stderr);
      expect(err).not.toContain("error:");
      expect(err).not.toContain("panic:");
      const out = await Bun.readableStreamToText(stdout);
      const first = out.slice(0, out.indexOf("\n"));
      expect(first).toEqual(expect.stringContaining("bun outdated "));
      expect(first).toEqual(expect.stringContaining("v1."));
      const rest = out.slice(out.indexOf("\n") + 1);
      expect(rest).toMatchSnapshot();
    });
  }
  test("in workspace", async () => {
    const { packageJson, packageDir } = await registry.createTestDir();
    await Promise.all([
      write(
        packageJson,
        JSON.stringify({
          name: "foo",
          workspaces: ["pkg1"],
          dependencies: {
            "no-deps": "1.0.0",
          },
        }),
      ),
      write(
        join(packageDir, "pkg1", "package.json"),
        JSON.stringify({
          name: "pkg1",
          dependencies: {
            "a-dep": "1.0.1",
          },
        }),
      ),
    ]);

    await runBunInstall(env, packageDir);

    let { stdout, stderr, exited } = spawn({
      cmd: [bunExe(), "outdated"],
      cwd: join(packageDir, "pkg1"),
      stdout: "pipe",
      stderr: "pipe",
      env,
    });

    const err = await Bun.readableStreamToText(stderr);
    expect(err).not.toContain("error:");
    expect(err).not.toContain("panic:");
    let out = await Bun.readableStreamToText(stdout);
    expect(out).toContain("a-dep");
    expect(out).not.toContain("no-deps");
    expect(await exited).toBe(0);

    ({ stdout, stderr, exited } = spawn({
      cmd: [bunExe(), "outdated"],
      cwd: packageDir,
      stdout: "pipe",
      stderr: "pipe",
      env,
    }));

    const err2 = await Bun.readableStreamToText(stderr);
    expect(err2).not.toContain("error:");
    expect(err2).not.toContain("panic:");
    let out2 = await Bun.readableStreamToText(stdout);
    expect(out2).toContain("no-deps");
    expect(out2).not.toContain("a-dep");
    expect(await exited).toBe(0);
  });

  test("NO_COLOR works", async () => {
    const { packageJson, packageDir } = await registry.createTestDir();
    await write(
      packageJson,
      JSON.stringify({
        name: "foo",
        dependencies: {
          "a-dep": "1.0.1",
        },
      }),
    );

    await runBunInstall(env, packageDir);

    const testEnv = { ...env, NO_COLOR: "1" };
    const { stdout, stderr, exited } = spawn({
      cmd: [bunExe(), "outdated"],
      cwd: packageDir,
      stdout: "pipe",
      stderr: "pipe",
      env: testEnv,
    });

    const err = await Bun.readableStreamToText(stderr);
    expect(err).not.toContain("error:");
    expect(err).not.toContain("panic:");

    const out = await Bun.readableStreamToText(stdout);
    expect(out).toContain("a-dep");
    const first = out.slice(0, out.indexOf("\n"));
    expect(first).toEqual(expect.stringContaining("bun outdated "));
    expect(first).toEqual(expect.stringContaining("v1."));
    const rest = out.slice(out.indexOf("\n") + 1);
    expect(rest).toMatchSnapshot();

    expect(await exited).toBe(0);
  });

  async function setupWorkspace(packageJson: string, packageDir: string) {
    await Promise.all([
      write(
        packageJson,
        JSON.stringify({
          name: "foo",
          workspaces: ["packages/*"],
          dependencies: {
            "no-deps": "1.0.0",
          },
        }),
      ),
      write(
        join(packageDir, "packages", "pkg1", "package.json"),
        JSON.stringify({
          name: "pkg1",
          dependencies: {
            "a-dep": "1.0.1",
          },
        }),
      ),
      write(
        join(packageDir, "packages", "pkg2", "package.json"),
        JSON.stringify({
          name: "pkg2222222222222",
          dependencies: {
            "prereleases-1": "1.0.0-future.1",
          },
        }),
      ),
    ]);
  }

  async function runBunOutdated(env: any, cwd: string, ...args: string[]): Promise<string> {
    const { stdout, stderr, exited } = spawn({
      cmd: [bunExe(), "outdated", ...args],
      cwd,
      stdout: "pipe",
      stderr: "pipe",
      env,
    });

    const err = await Bun.readableStreamToText(stderr);
    expect(err).not.toContain("error:");
    expect(err).not.toContain("panic:");
    const out = await Bun.readableStreamToText(stdout);
    const exitCode = await exited;
    expect(exitCode).toBe(0);
    return out;
  }

  test("--filter with workspace names and paths", async () => {
    const { packageJson, packageDir } = await registry.createTestDir();
    await setupWorkspace(packageJson, packageDir);
    await runBunInstall(env, packageDir);

    let out = await runBunOutdated(env, packageDir, "--filter", "*");
    expect(out).toContain("foo");
    expect(out).toContain("pkg1");
    expect(out).toContain("pkg2222222222222");

    out = await runBunOutdated(env, join(packageDir, "packages", "pkg1"), "--filter", "./");
    expect(out).toContain("pkg1");
    expect(out).not.toContain("foo");
    expect(out).not.toContain("pkg2222222222222");

    // in directory that isn't a workspace
    out = await runBunOutdated(env, join(packageDir, "packages"), "--filter", "./*", "--filter", "!pkg1");
    expect(out).toContain("pkg2222222222222");
    expect(out).not.toContain("pkg1");
    expect(out).not.toContain("foo");

    out = await runBunOutdated(env, join(packageDir, "packages", "pkg1"), "--filter", "../*");
    expect(out).not.toContain("foo");
    expect(out).toContain("pkg2222222222222");
    expect(out).toContain("pkg1");
  });

  test("dependency pattern args", async () => {
    const { packageJson, packageDir } = await registry.createTestDir();
    await setupWorkspace(packageJson, packageDir);
    await runBunInstall(env, packageDir);

    let out = await runBunOutdated(env, packageDir, "no-deps", "--filter", "*");
    expect(out).toContain("no-deps");
    expect(out).not.toContain("a-dep");
    expect(out).not.toContain("prerelease-1");

    out = await runBunOutdated(env, packageDir, "a-dep");
    expect(out).not.toContain("a-dep");
    expect(out).not.toContain("no-deps");
    expect(out).not.toContain("prerelease-1");

    out = await runBunOutdated(env, packageDir, "*", "--filter", "*");
    expect(out).toContain("no-deps");
    expect(out).toContain("a-dep");
    expect(out).toContain("prereleases-1");
  });

  test("scoped workspace names", async () => {
    const { packageJson, packageDir } = await registry.createTestDir();
    await Promise.all([
      write(
        packageJson,
        JSON.stringify({
          name: "@foo/bar",
          workspaces: ["packages/*"],
          dependencies: {
            "no-deps": "1.0.0",
          },
        }),
      ),
      write(
        join(packageDir, "packages", "pkg1", "package.json"),
        JSON.stringify({
          name: "@scope/pkg1",
          dependencies: {
            "a-dep": "1.0.1",
          },
        }),
      ),
    ]);

    await runBunInstall(env, packageDir);

    let out = await runBunOutdated(env, packageDir, "--filter", "*");
    expect(out).toContain("@foo/bar");
    expect(out).toContain("@scope/pkg1");

    out = await runBunOutdated(env, packageDir, "--filter", "*", "--filter", "!@foo/*");
    expect(out).not.toContain("@foo/bar");
    expect(out).toContain("@scope/pkg1");
  });

  test("catalog dependencies", async () => {
    const { packageJson, packageDir } = await registry.createTestDir();
    await Promise.all([
      write(
        packageJson,
        JSON.stringify({
          name: "catalog-outdated-test",
          workspaces: {
            packages: ["packages/*"],
            catalog: {
              "no-deps": "1.0.0",
            },
            catalogs: {
              dev: {
                "a-dep": "1.0.1",
              },
            },
          },
        }),
      ),
      write(
        join(packageDir, "packages", "pkg1", "package.json"),
        JSON.stringify({
          name: "pkg1",
          dependencies: {
            "no-deps": "catalog:",
          },
          devDependencies: {
            "a-dep": "catalog:dev",
          },
        }),
      ),
    ]);

    await runBunInstall(env, packageDir);

    const out = await runBunOutdated(env, packageDir, "--filter", "*");
    expect(out).toContain("no-deps");
    expect(out).toContain("a-dep");
  });
});
