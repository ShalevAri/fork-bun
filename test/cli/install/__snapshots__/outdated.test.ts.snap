// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`outdated normal dep, smaller than column title 1`] = `
"┌──────────┬─────────┬────────┬────────┐
│ \x1B[1m\x1B[34mPackage\x1B[0m  │ \x1B[1m\x1B[34mCurrent\x1B[0m │ \x1B[1m\x1B[34mUpdate\x1B[0m │ \x1B[1m\x1B[34mLatest\x1B[0m │
├──────────┼─────────┼────────┼────────┤
│ no-deps\x1B[2m\x1B[0m  │ 1.0.0   │ \x1B[2m1.0.0\x1B[0m  │ \x1B[0m\x1B[1m\x1B[31m2.0.0\x1B[0m  │
└──────────┴─────────┴────────┴────────┘
"
`;

exports[`outdated normal dep, larger than column title 1`] = `
"┌───────────────┬────────────────┬────────────────┬────────────────┐
│ \x1B[1m\x1B[34mPackage\x1B[0m       │ \x1B[1m\x1B[34mCurrent\x1B[0m        │ \x1B[1m\x1B[34mUpdate\x1B[0m         │ \x1B[1m\x1B[34mLatest\x1B[0m         │
├───────────────┼────────────────┼────────────────┼────────────────┤
│ prereleases-1\x1B[2m\x1B[0m │ 1.0.0-future.1 │ \x1B[2m1.0.0-future.1\x1B[0m │ \x1B[2m1.0.0-future.\x1B[0m\x1B[1m\x1B[31m4\x1B[0m │
└───────────────┴────────────────┴────────────────┴────────────────┘
"
`;

exports[`outdated dev dep, smaller than column title 1`] = `
"┌───────────────┬─────────┬────────┬────────┐
│ \x1B[1m\x1B[34mPackage\x1B[0m       │ \x1B[1m\x1B[34mCurrent\x1B[0m │ \x1B[1m\x1B[34mUpdate\x1B[0m │ \x1B[1m\x1B[34mLatest\x1B[0m │
├───────────────┼─────────┼────────┼────────┤
│ no-deps\x1B[2m (dev)\x1B[0m │ 1.0.0   │ \x1B[2m1.0.0\x1B[0m  │ \x1B[0m\x1B[1m\x1B[31m2.0.0\x1B[0m  │
└───────────────┴─────────┴────────┴────────┘
"
`;

exports[`outdated dev dep, larger than column title 1`] = `
"┌─────────────────────┬────────────────┬────────────────┬────────────────┐
│ \x1B[1m\x1B[34mPackage\x1B[0m             │ \x1B[1m\x1B[34mCurrent\x1B[0m        │ \x1B[1m\x1B[34mUpdate\x1B[0m         │ \x1B[1m\x1B[34mLatest\x1B[0m         │
├─────────────────────┼────────────────┼────────────────┼────────────────┤
│ prereleases-1\x1B[2m (dev)\x1B[0m │ 1.0.0-future.1 │ \x1B[2m1.0.0-future.1\x1B[0m │ \x1B[2m1.0.0-future.\x1B[0m\x1B[1m\x1B[31m4\x1B[0m │
└─────────────────────┴────────────────┴────────────────┴────────────────┘
"
`;

exports[`outdated peer dep, smaller than column title 1`] = `
"┌────────────────┬─────────┬────────┬────────┐
│ \x1B[1m\x1B[34mPackage\x1B[0m        │ \x1B[1m\x1B[34mCurrent\x1B[0m │ \x1B[1m\x1B[34mUpdate\x1B[0m │ \x1B[1m\x1B[34mLatest\x1B[0m │
├────────────────┼─────────┼────────┼────────┤
│ no-deps\x1B[2m (peer)\x1B[0m │ 1.0.0   │ \x1B[2m1.0.0\x1B[0m  │ \x1B[0m\x1B[1m\x1B[31m2.0.0\x1B[0m  │
└────────────────┴─────────┴────────┴────────┘
"
`;

exports[`outdated peer dep, larger than column title 1`] = `
"┌──────────────────────┬────────────────┬────────────────┬────────────────┐
│ \x1B[1m\x1B[34mPackage\x1B[0m              │ \x1B[1m\x1B[34mCurrent\x1B[0m        │ \x1B[1m\x1B[34mUpdate\x1B[0m         │ \x1B[1m\x1B[34mLatest\x1B[0m         │
├──────────────────────┼────────────────┼────────────────┼────────────────┤
│ prereleases-1\x1B[2m (peer)\x1B[0m │ 1.0.0-future.1 │ \x1B[2m1.0.0-future.1\x1B[0m │ \x1B[2m1.0.0-future.\x1B[0m\x1B[1m\x1B[31m4\x1B[0m │
└──────────────────────┴────────────────┴────────────────┴────────────────┘
"
`;

exports[`outdated optional dep, smaller than column title 1`] = `
"┌────────────────────┬─────────┬────────┬────────┐
│ \x1B[1m\x1B[34mPackage\x1B[0m            │ \x1B[1m\x1B[34mCurrent\x1B[0m │ \x1B[1m\x1B[34mUpdate\x1B[0m │ \x1B[1m\x1B[34mLatest\x1B[0m │
├────────────────────┼─────────┼────────┼────────┤
│ no-deps\x1B[2m (optional)\x1B[0m │ 1.0.0   │ \x1B[2m1.0.0\x1B[0m  │ \x1B[0m\x1B[1m\x1B[31m2.0.0\x1B[0m  │
└────────────────────┴─────────┴────────┴────────┘
"
`;

exports[`outdated optional dep, larger than column title 1`] = `
"┌──────────────────────────┬────────────────┬────────────────┬────────────────┐
│ \x1B[1m\x1B[34mPackage\x1B[0m                  │ \x1B[1m\x1B[34mCurrent\x1B[0m        │ \x1B[1m\x1B[34mUpdate\x1B[0m         │ \x1B[1m\x1B[34mLatest\x1B[0m         │
├──────────────────────────┼────────────────┼────────────────┼────────────────┤
│ prereleases-1\x1B[2m (optional)\x1B[0m │ 1.0.0-future.1 │ \x1B[2m1.0.0-future.1\x1B[0m │ \x1B[2m1.0.0-future.\x1B[0m\x1B[1m\x1B[31m4\x1B[0m │
└──────────────────────────┴────────────────┴────────────────┴────────────────┘
"
`;

exports[`outdated NO_COLOR works 1`] = `
"|--------------------------------------|
| Package  | Current | Update | Latest |
|----------|---------|--------|--------|
| a-dep    | 1.0.1   | 1.0.1  | 1.0.10 |
|--------------------------------------|
"
`;
